lib/model/CResourceMonitor.cc

```diff
@@ -306,6 +305,25 @@ void CResourceMonitor::clearExtraMemory() {
     }
 }
 
+void CResourceMonitor::decreaseMargin(core_t::TTime elapsedTime) {
+    // We choose to increase the margin to close to 1 on the order
+    // time it takes to detect diurnal periodic components. These
+    // will be the overwhelmingly common source of additional memory
+    // so the model memory should be accurate (on average) in this
+    // time frame.
+    double scale{1.0 - static_cast<double>(elapsedTime) /
```

Comment:
It seems the scale is fixed for a given bucket span. Should we consider setting it on construction?

Reply:
We could do given current usage. Although in the current usage this is called at the bucket frequency this doesn't feel like it has to be the case and we'd lose the ability to do this if that changed. Also, I think if we move this away from the API to adjust the margin it hides an important part of the implementation from the caller. If I changed how this function is called I think it would be easy to overlook that I need to change the value computed in the constructor. I think on balance I prefer to keep as is for that reason. What do you think?

Reply:
I agree. This gives us flexibility to change the way it works if needed in the future.